SampleApp/app/src/main/java/com/placed/android/sampleapp/MainActivity.java

```diff
@@ -55,24 +54,42 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
     }
 
     private void registerUser() {
-        if (!PreferenceManager.getDefaultSharedPreferences(this).getBoolean(PREF_KEY_PLACED_DIALOG_SHOWN, false)) {
-            PreferenceManager.getDefaultSharedPreferences(this)
-                    .edit()
-                    .putBoolean(PREF_KEY_PLACED_DIALOG_SHOWN, true)
-                    .apply();
+        if (!PreferenceManager.getDefaultSharedPreferences(this).getBoolean(TERMS_AND_PRIVACY_ACCEPTED_KEY, false)) {
+            SampleDialog sampleDialog = new SampleDialog(this);
+            sampleDialog.show();
+            sampleDialog.onActionListener = this;
+        }
+        else {
+            PlacedAgent.registerUser(MainActivity.this);
+        }
+    }
+
+    @Override
+    public void onActionReceived(SampleDialogAction action) {
+        switch (action) {
+            case TERMS: {
+                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.placed.com/terms-of-service"));
+                startActivity(browserIntent);
+            }
+            break;
+
+            case PRIVACY: {
+                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.placed.com/privacy-policy"));
+                startActivity(browserIntent);
+            }
```

Comment:
Nitpick: I don't think the `{` and `}` in a `case` statement is strictly necessary.